pkg/disk/lsblk_issue_1107_test.go

```diff
@@ -0,0 +1,227 @@
+		oldDeviceTypeFunc := func(dt string) bool {
+			return dt == "disk" || dt == "lvm" || dt == "part" // No RAID support!
+		}
+
+		raidDevice := &BlockDevice{
+			Name:       "/dev/md127",
+			Type:       "raid1",
+			FSType:     "ext4",
+			MountPoint: "/",
+		}
+
+		op := &Op{}
+		require.NoError(t, op.applyOpts([]OpOption{
+			WithMountPoint(DefaultMountPointFunc),
+			WithDeviceType(oldDeviceTypeFunc), // OLD function without RAID
+			WithFstype(DefaultFsTypeFunc),
+		}))
+
+		fstypeCache := make(map[string]string)
+		result := processBlockDevice(ctx, raidDevice, "", 0, op, fstypeCache)
+
+		// This FAILS because old function doesn't support RAID
+		assert.False(t, result, "RAID device should be rejected by old function")
+	})
+}
+
+// TestIssue1107_FullFixValidation validates the complete fix using the actual test data structure
+func TestIssue1107_FullFixValidation(t *testing.T) {
+	ctx := context.Background()
+
+	// Simulate the actual problematic structure from lsblk.withlv.censored.json
+	// that was failing with the "no block device found" error
+	problematicStructure := []BlockDevice{
+		// Problem 1: Nested LVM (needs recursive processing)
+		{
+			Name:       "/dev/nvme2n1",
+			Type:       "disk",
+			FSType:     "LVM2_member",
+			MountPoint: "",
+			Children: []BlockDevice{
+				{
+					Name:       "/dev/mapper/vg_nvme-lv_nvme_rimage_1",
+					Type:       "lvm",
+					FSType:     "",
+					MountPoint: "",
+					Children: []BlockDevice{
+						{
+							Name:       "/dev/mapper/vg_nvme-lv_nvme",
+							Type:       "lvm",
+							FSType:     "ext4",
+							MountPoint: "/mnt/local_disk",
+						},
+					},
+				},
+			},
+		},
+		// Problem 2: RAID device (needs RAID support in filter)
+		{
+			Name:       "/dev/sda",
+			Type:       "disk",
+			FSType:     "",
+			MountPoint: "",
+			Children: []BlockDevice{
+				{
+					Name:       "/dev/sda2",
+					Type:       "part",
+					FSType:     "linux_raid_member",
+					MountPoint: "",
+					Children: []BlockDevice{
+						{
+							Name:       "/dev/md127",
+							Type:       "raid1",
+							FSType:     "ext4",
+							MountPoint: "/",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	op := &Op{}
+	require.NoError(t, op.applyOpts([]OpOption{
+		WithMountPoint(DefaultMountPointFunc),
+		WithDeviceType(DefaultDeviceTypeFunc),
+		WithFstype(DefaultFsTypeFunc),
+	}))
+
+	fstypeCache := make(map[string]string)
+	foundDevices := 0
+
+	for i := range problematicStructure {
+		dev := &problematicStructure[i]
+		if processBlockDevice(ctx, dev, "", 0, op, fstypeCache) {
+			foundDevices++
+		}
+	}
+
+	// Both devices should be found with the complete fix
+	assert.Equal(t, 2, foundDevices, "both problematic devices should be found with complete fix")
+	assert.NotZero(t, foundDevices, "should NOT trigger 'no block device found' error")
+}
+
+// TestIssue1107_DocumentWhyBothChangesNeeded documents the necessity of both changes
+func TestIssue1107_DocumentWhyBothChangesNeeded(t *testing.T) {
```

Comment:
Why do we need such a test function?  we can add comments or write a detailed git message.

Reply:
Oh, forgot to implement this... removing